lib/datadog.ts

```diff
@@ -8,36 +8,88 @@
+import { applyLayers, redirectHandlers, addForwarder, applyEnvVariables, defaultEnvVar } from "./index";
 
 export interface DatadogProps {
-  lambdaFunctions: lambda.Function[];
   pythonLayerVersion?: number;
   nodeLayerVersion?: number;
   addLayers?: boolean;
   forwarderARN?: string;
+  flushMetricsToLogs?: boolean;
+  site?: string;
+  apiKey?: string;
+  apiKMSKey?: string;
+  enableDDTracing?: boolean;
+  injectLogContext?: boolean;
 }
 
+export const site_list: string[] = [
+  "datadoghq.com",
+  "datadoghq.eu",
+  "us3.datadoghq.com",
+  "ddog-gov.com"
+]
+
 export class Datadog extends cdk.Construct {
-  /** allows accessing the counter function */
+  scope: cdk.Construct;
+  props: DatadogProps
   constructor(scope: cdk.Construct, id: string, props: DatadogProps) {
     super(scope, id);
-    if (props.addLayers === undefined) {
-      props.addLayers = true;
+    this.scope = scope;
+    this.props = props;
+  }
+
+  public addLambdaFunctions(lambdaFunctions: lambda.Function[]){
+    if (this.props.addLayers === undefined) {
+      this.props.addLayers = defaultEnvVar.addLayers;
+    }
+    if (this.props.flushMetricsToLogs === undefined) {
+      this.props.flushMetricsToLogs = defaultEnvVar.flushMetricsToLogs;
+    }
+    if (this.props.flushMetricsToLogs === false && this.props.site === undefined) {
+      throw new Error('A site is required if flushMetricsToLogs is set to false.');
+    }
+    if (this.props.site === undefined) {
+      this.props.site = defaultEnvVar.site;
+    }
+    if (!(site_list.includes(this.props.site))) {
+      throw new Error('Invalid site URL. Must be either datadoghq.com, datadoghq.eu, us3.datadoghq.com, or ddog-gov.com.')
+    }
+    if ((this.props.apiKey != undefined && this.props.apiKMSKey != undefined && this.props.flushMetricsToLogs === false) ||
+    (this.props.apiKey === undefined && this.props.apiKMSKey === undefined && this.props.flushMetricsToLogs === false)) {
+      throw new Error('The parameters apiKey and apiKMSKey are mutually exclusive. Please set one or the other but not both if flushMetricsToLogs is set to false.')
+    };
+    if (this.props.enableDDTracing === undefined) {
+      this.props.enableDDTracing = defaultEnvVar.enableDDTracing;
+    }
+    if ((this.props.enableDDTracing === true || this.props.enableDDTracing === undefined) && this.props.forwarderARN === undefined) {
+      throw new Error('A forwarderARN of the Datadog forwarder lambda function is required for Datadog Tracing (enabled by default). This can be disabled by setting enableDDTracing: false.')
+    }
+    if (this.props.injectLogContext === undefined) {
+      this.props.injectLogContext = defaultEnvVar.injectLogContext;
     }
-    if (props != undefined && props.lambdaFunctions.length > 0) {
-      const region = `${props.lambdaFunctions[0].env.region}`;
+    if (this.props != undefined && lambdaFunctions.length > 0) {
+      const region = `${lambdaFunctions[0].env.region}`;
```

Comment:
Are all of the functions guaranteed to be in the same region? I'm 99% sure adding layers will break if they aren't. Do we want to check for that?